lib/firebase.dart

```diff
@@ -76,13 +75,17 @@ void _fbAuthChanged(
 }
 
 // Read data
-Future<model.Config> fetchConfig() async {
+Future<Map<String, dynamic>> fetchConfig() async {
   logger.debug('Fetching config from firebase ..');
   var chartsConfigRef = firebase.firestore().doc(fb_constants.metadataPath);
   var configSnapshot = await chartsConfigRef.get();
-  var configData = configSnapshot.data();
+  return configSnapshot.data();
+}
 
-  return model.Config.fromData(configData);
+Future<Null> updateConfig(Map<String, dynamic> data) async {
+  var chartsConfigRef = firebase.firestore().doc(fb_constants.metadataPath);
+  var configSnap = await chartsConfigRef.update(data: data);
+  return configSnap;
```

Comment:
We might need to look at catching errors in the places where we talk to firebase, and display them in the UI. Doesn't happen often, and primarily on bad networks, so it's not urgent, just something to add to the todolist :)

Reply:
Added a try catch for now to show an alert, but will go through all the firebase functions to make sure they show the errors. Tracking in here: #19